tracer/src/Datadog.Trace/Span.cs

```diff
@@ -77,7 +78,16 @@ internal Span(SpanContext context, DateTimeOffset? start, ITags tags)
         internal string ServiceName
         {
             get => Context.ServiceNameInternal;
-            set => Context.ServiceNameInternal = value;
+            set
+            {
+                if (_baseService is null && !string.Equals(value, Context.ServiceNameInternal))
```

Comment:
No actual use to keep `_baseService` as a variable, as it is not compared to anything other than null or retrieved ever again. But I didn't want to use GetTag because it iterates over every tag. Thoughts!

Reply:
It looks like you could use a `bool` to indicate if the tag was already added, but I'm not sure it makes any real difference:

```csharp
private bool _baseServiceTagSet;

// ...

if (!_baseServiceTagSet && ...
{
    _baseServiceTagSet = true;
   // ...
```